test/pkg_finish_test.rb

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+require "packager/clients/pkg_finish"
+require "packages/repository"
+require "uri"
+require "tmpdir"
+require "fileutils"
+
+describe Yast::PkgFinishClient do
+  Yast.import "Pkg"
+  Yast.import "Installation"
+  Yast.import "WFM"
+
+  FAILED_PKGS_PATH = "/var/lib/YaST2/failed_packages"
+
+  subject(:client) { Yast::PkgFinishClient.new }
+  let(:repositories) { [] }
+
+  before do
+    allow(Yast::WFM).to receive(:Args).and_return(args)
+    allow(::Packages::Repository).to receive(:enabled).and_return(repositories)
+  end
+
+  describe "Info" do
+    let(:args) { ["Info"] }
+
+    it "returns a hash describing the client" do
+      allow(client).to receive(:_).and_return("title")
+      expect(client.run).to eq({
+          "steps" => 1,
+          "title" => "title",
+          "when" => [:installation, :update, :autoinst]})
+    end
+  end
+
+  describe "Write" do
+    let(:args) { ["Write"] }
+    let(:destdir) { "/mnt" }
+    let(:update) { false }
+
+    before do
+      allow(Yast::Installation).to receive(:destdir).and_return(destdir)
+      allow(Yast::Mode).to receive(:update).and_return(update)
+      allow(Yast::Stage).to receive(:initial).and_return(true)
+      allow(Yast::Pkg).to receive(:SourceLoad)
+    end
+
+    it "saves repository information" do
+      expect(Yast::Pkg).to receive(:SourceLoad)
+      expect(Yast::Pkg).to receive(:SourceSaveAll)
+      expect(Yast::Pkg).to receive(:TargetFinish)
+      expect(Yast::Pkg).to receive(:SourceCacheCopyTo).with(destdir)
+      allow(Yast::WFM).to receive(:Execute)
+      expect(client.run).to be_nil
+    end
+
+    it "copies failed_packages list under destination dir" do
+      stub_const("Yast::Pkg", double("pkg").as_null_object)
+      allow(::File).to receive(:exist?).and_call_original
+      expect(::File).to receive(:exist?).with(FAILED_PKGS_PATH)
+        .and_return(true)
+      expect(::FileUtils).to receive(:cp)
```

Comment:
The same here, moreover a plain `FileUtils` is used below so it is even inconsistent...